samples/Azure.ResourceManager.Sample/Generated/AvailabilitySetCollection.cs

```diff
@@ -65,6 +65,10 @@ internal static void ValidateResourceId(ResourceIdentifier id)
         /// <term>Operation Id</term>
         /// <description>AvailabilitySets_CreateOrUpdate</description>
         /// </item>
+        /// <item>
+        /// <term>Resource Type</term>
+        /// <description>Microsoft.Compute/availabilitySets</description>
```

Comment:
I think our customers might get confused why an operation has a "resource type".
More useful information for our customers might be "the default api-version" of this method which makes total sense to our customers.
And since this PR is trying to "resolve" issue https://github.com/Azure/autorest.csharp/issues/3650 which comes from an inquiry of "how to set the api-version of an operation". If the content of this PR is only showing up "resource type" of an operation, I cannot admit that this PR **resolves** the issue.
This PR does provide more information, but does not really resolve the issue.

Reply:
As I said, the "Resource Type" is a general information related to the operation, I am fine to improve it to something like "The Resource Type providing the Operation" or something like that. but i dont think we should provide the information explicitly for set-apiversion api which is just one kind of usage of that information and also each api should decide their own behavior by themselves. Other api shouldn't comment on how set-apiversion API would work or use what kind of information, otherwise we are adding unnecessary coupling and making it harder to maintain both APIs in the future.

As for the issue #3650, this is like the first step of the fix. i am fine to keep it open until we add more comment to the set-apiVersion API when the "Resource Type" is available in most important sdk.